File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* s, const char* skip, int index) {
    bool A[26];
    for(int i=0;i<26;i++){A[i] = true;}

    int skip_len=0;
    for(skip_len=0;skip[skip_len];skip_len++){A[skip[skip_len]-'a'] = false;}

    int count = 26 - skip_len;
    int a[26];
    int b[26];
    int j=0;
    for(int i=0;i<26;i++){
        if(A[i]){a[j] = i;b[i]=j;j++;}
    }


    int len=0;
    while(s[len]){len++;}    
    char* answer = (char*)malloc(len*sizeof(char)+1);
    answer[len] = '\0';
    for(int i=0;i<len;i++){
        

        answer[i] = 'a' + (a[(b[s[i]-'a']+index)%count]);
    }


    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호 - Python/둘만의 암호.py

def solution(s, skip, index):
    answer = ''
    not_skip = [True for i in range(26)]
    for skip_char in skip:
        not_skip[ord(skip_char) - 97] = False

    for c in s:
        count = 0
        C_i = ord(c) - 97
        while count < index:
            C_i = (C_i + 1) % 26
            if not_skip[C_i]:
                count += 1

        answer += chr(C_i + 97)
            
        
    return answer